packages/node_modules/@ciscospark/internal-plugin-reaction/src/reaction.js

```diff
@@ -0,0 +1,117 @@
+/*!
+ * Copyright (c) 2015-2017 Cisco Systems, Inc. See LICENSE file.
+ */
+
+import {SparkPlugin} from '@ciscospark/spark-core';
+import uuid from 'uuid';
+
+const Reaction = SparkPlugin.extend({
+  namespace: 'Reaction',
+
+  /**
+  * helper function to create the request object
+  * @param {Object} options
+  * @returns {Object} the reaction request object
+  */
+  createReactionRequest(options) {
```

Comment:
Take a look at the [rooms](https://github.com/ciscospark/spark-js-sdk/blob/master/packages/node_modules/@ciscospark/plugin-rooms/src/rooms.js) plugin, specifically how the `RoomObject` is defined, it might be worthwhile to copy that method of passing options.